### PR TITLE
Add UI-only database configuration method for extended authorization scenarios

### PR DESCRIPTION
**Pull request by willbarrett**
### CATEGORY

Choose one

- [ ] Bug Fix
- [X] Enhancement (new features, refinement)
- [ ] Refactor
- [ ] Add tests
- [ ] Build / Development Environment
- [ ] Documentation

### SUMMARY

Problem statement: currently connections to BigQuery databases cannot be configured without adding code to Superset or adding configuration to its execution environment in the form of Google Cloud account credentials.

Solution proposed: Add an encrypted field on the `dbs` table to hold extended connection information. This field can hold JSON configuration including secrets and add that information on to the database connection object during instantiation. This allows us to configure a service account in Google Cloud and connect to a BigQuery dataset without any code or configuration changes in Superset.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

Screen Shot 2019-10-24 at 12 50 46 PM

### TEST PLAN

* Create a dataset in BigQuery in Google Cloud
* Create a service account in Google Cloud with access to the dataset
* Download the service account's identity file in JSON format
* Open the `Add Database` screen in Superset 
* Fill in the SqlAlchemy URL with `bigquery://project-name` where `project-name` is the name of the Google Cloud project that your BigQuery dataset is associated with.
* Add JSON to the `Secure Extra` field with the following structure:
```json
{
    "credentials_info": <content from the identity file from Google Cloud>
}
```
* Click "Test Connection" - It should say the connection is OK.

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [X] Changes UI
- [X] Requires DB Migration.
- [X] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### REVIEWERS
@mistercrunch @dpgaspar

### CODE REVIEW

File: superset/views/database/__init__.py [hunk 1]
```diff
@@ -94,6 +94,7 @@ class DatabaseMixin:
         "impersonate_user",
         "allow_multi_schema_metadata_fetch",
         "extra",
+        "encrypted_extra",
```
Comment (mistercrunch) [inline superset/views/database/__init__.py]: I just confirmed that this doesn't show in `show_columns`

**Review by mistercrunch — APPROVED** [review]

**Review by mistercrunch — APPROVED** [review]

File: superset/models/core.py [hunk 2]
```diff
@@ -748,6 +748,9 @@ class Database(Model, AuditMixinNullable, ImportMixin):
     """
         ),
     )
+    encrypted_extra = Column(
+        EncryptedType(Text, config.get("SECRET_KEY")), nullable=True
+    )
```
Comment (villebro) [inline superset/models/core.py]: Nit: as per #8409, config variables are assumed to always be set. So this should be changed to ...`config["SECRET_KEY"]`

**Review by villebro — APPROVED** [review]
This is a very nice feature 👍 One small nit, other than that LGTM.

File: superset/views/database/__init__.py [hunk 3]
```diff
@@ -203,6 +211,7 @@ class DatabaseMixin:
         "sqlalchemy_uri": _("SQLAlchemy URI"),
         "cache_timeout": _("Chart Cache Timeout"),
         "extra": _("Extra"),
+        "encrypted_extra": ("Secure Extra"),
```
Comment (dpgaspar) [inline superset/views/database/__init__.py]: `"encrypted_extra": _("Secure Extra"),` So it gets translated (eventually)

**Review by dpgaspar — APPROVED** [review]

**Pull request merged**